ConverterViewController.swift

```diff
@@ -0,0 +1,174 @@
+let FIELDMAXLENGTH = 8
+let DEFAULTAMOUNT = 1.0
+
+class ConverterViewController: UIViewController, UITextFieldDelegate {
+    
+    var _dollarAmount: Double? = DEFAULTAMOUNT
+    var _exchangesDictionary:[String:Double]?
+    let _currencyFormatter : NSNumberFormatter = NSNumberFormatter()
+    var _requestedWasOk = false
+    
+    @IBOutlet weak var lblGBP: UILabel!
+    @IBOutlet weak var lblEUR: UILabel!
+    @IBOutlet weak var lblBRL: UILabel!
+    @IBOutlet weak var lblJPY: UILabel!
+    
+    @IBOutlet weak var dollarNumberTextField: UITextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //sets format number style to currency
+        _currencyFormatter.numberStyle = NSNumberFormatterStyle.CurrencyStyle
+        
+        //set delegate for dollarnumberUITextField
+        self.dollarNumberTextField.delegate = self
+        
+        //get current dollar price from fixer.io
+        getCurrencyValues(self.updateValues)
+        
+        // Gesture that recognices tap to hide keyboard
+        let tapGesture = UITapGestureRecognizer(target: self, action:Selector("endEditing:"))
+        tapGesture.cancelsTouchesInView = false
+        self.view.addGestureRecognizer(tapGesture);
+    }
+    
+    // fetches exchange values for dollar using a get request and receiving a callback
+    func getCurrencyValues(completion: (() -> Void)!) {
+        
+        // Create NSURL Ibject
+        let pricesUrl = NSURL(string: EXCHANGESURL);
+        
+        // Creaste URL Request
+        let pricesRequest = NSMutableURLRequest(URL:pricesUrl!);
+        
+        // Set request HTTP method to GET. It could be POST as well
+        pricesRequest.HTTPMethod = REQUESTMETHOD
+        
+        // Excute HTTP Request
+        let task = NSURLSession.sharedSession().dataTaskWithRequest(pricesRequest) {
+            data, response, error in
+            
+            // Check for error
+            if error != nil
+            {
+                print("error=\(error)")
+                return
+            }
+            
+            // Print out response string
+            let responseString = NSString(data: data!, encoding: NSUTF8StringEncoding)
+            print("responseString = \(responseString)")
+            
+            
+            // Convert server json response to NSDictionary
+            do {
+                if let convertedJsonIntoDict = try NSJSONSerialization.JSONObjectWithData(data!, options: []) as? NSDictionary {
+                    
+                    self._requestedWasOk = true
+                    
+                    self._exchangesDictionary = convertedJsonIntoDict["rates"] as? [String : Double]
+                    
+                    completion() //callback
+                    
+                }
+            } catch let error as NSError {
+                print(error.localizedDescription)
+                
+                self._requestedWasOk = false
+            }
```

Comment:
You are not calling the completion block if there's an error, which may lead to the UI waiting forever for a response that never came. In this cases what we usually do is to have a closure that passes the expected result and an error. Both are optional so we can pass `nil` if needed